File: H2OApp/app/src/main/java/com/gs/h2oapp/Entity/Coupon.java
package com.gs.h2oapp.Entity;

import com.google.gson.annotations.SerializedName;

import java.math.BigDecimal;

/**
 * Created by Warley Lima 
 */
public class Coupon {
    private int idC;
    @SerializedName("vc")
    private String cupom;
    @SerializedName("dti")
    private String dateI;
    @SerializedName("dtf")
    private String dateF;
    @SerializedName("vd")
    private BigDecimal desconto;

    public Coupon() {
    }

    public int getIdC() {
        return idC;
    }

    public void setIdC(int idC) {
        this.idC = idC;
    }

    public String getCupom() {
        return cupom;
    }

    public void setCupom(String cupom) {
        this.cupom = cupom;
    }

    public String getDateI() {
        return dateI;
    }

    public void setDateI(String dateI) {
        this.dateI = dateI;
    }

    public String getDateF() {
        return dateF;
    }

    public void setDateF(String dateF) {
        this.dateF = dateF;
    }

    public BigDecimal getDesconto() {
        return desconto;
    }

    public void setDesconto(BigDecimal desconto) {
        this.desconto = desconto;
    }
}

File: H2OApp-WEB/src/app/shared/Model/coupon.ts
export class Coupon {
  public vc: string;
  public dti: string;
  public dtf: string;
  public vd: string;
  public idC: number;
   constructor(idCupom: number, valor: string,  dti: string, dtf: string, vd: string) {
     this.vc = valor;
     this.idC = idCupom;
     this.dti = dti;
     this.dtf = dtf;
     this.vd = vd;
   }
}
